core/src/main/scala/org/apache/spark/scheduler/SparkListenerBus.scala

```diff
@@ -24,7 +24,13 @@ import org.apache.spark.util.ListenerBus
  */
 private[spark] trait SparkListenerBus extends ListenerBus[SparkListener, SparkListenerEvent] {
 
+  private[spark] var filter: DefaultSparkListenerEventFilter = new DefaultSparkListenerEventFilter
+
   override def onPostEvent(listener: SparkListener, event: SparkListenerEvent): Unit = {
+    if (!filter.validate(event)) {
+      return  
```

Comment:
I think the `DefaultSparkListenerEventFilter` is probably adding an abstraction without any really good reason.  If we do stick w/ the new `SparkListenerBlockUpdate` events, I think its better if you just put the filter into the right method on the event logging listener and remove `DefaultSparkListenerEventFilter`.